### PR TITLE
Use zipper rather than jdk/jar in the thrift rules

### LINKED ISSUE #282: use bazel_tools zip instead of a call to jar in thrift rules
here:

https://github.com/bazelbuild/rules_scala/blob/master/thrift/thrift.bzl#L54

and we do a dance to try to avoid letting timestamps in.

But in fact, bazel tools has a tool that does this:
https://github.com/bazelbuild/bazel/blob/master/tools/zip/BUILD#L9

by using zipper, it should get cleaner and less likely to have weird timestamp issues.

### PR DESCRIPTION
**Pull request by johnynek**
Bazel has a built in command for building zips deterministically. Use that rather than the find + timestamp setting trick.

closes #282

### CODE REVIEW

File: thrift/thrift.bzl [hunk 1]
```diff
@@ -135,8 +148,7 @@ thrift_library = rule(
       "absolute_prefixes": attr.string_list(),
       # This is a list of JARs which only contain Thrift files
       "external_jars": attr.label_list(),
-      "_jar": attr.label(executable=True, cfg="host", default=Label("@bazel_tools//tools/jdk:jar"), allow_files=True),
-      "_jdk": attr.label(default=Label("//tools/defaults:jdk"), allow_files=True),
+      "_zipper": attr.label(executable=True, cfg="host", default=Label("@bazel_tools//tools/zip:zipper"), allow_files=True)
```
Comment (softprops) [inline thrift/thrift.bzl]: TIL this exsited! This kind of thing needs better publicity. I've cargo culted around the restamped archive code in like every archive rule I've written

Comment (johnynek) [reply thrift/thrift.bzl]: right?

**Comment by johnynek** [comment]
I tested this final version on our largest internal repo and it seemed to work (and get rid of an annoying warning we saw when we had targets with no thrifts that only had external jars).

**Comment by ittaiz** [comment]
@johnynek looks good.
A) Can you squash the 6 commits which are related to this so I'll be able to rebase? I think I can't squash due to the shared commits and I'd rather not add all these intermediate commits.
B) Shouldn't we use the zipper also in the `scala.bzl`? I seem to remember we also do a song and dance there.

**Comment by johnynek** [comment]
@ittaiz if we squash, then the 2 branches that depend on this will likely have a painful merge process. I would really like to just regular merge these as is to avoid that. Then the other two can be easily squash merged into the repo saving precious time.

**Comment by johnynek** [comment]
oh, wait, you did merge regular, so I think I can squash merge this now into one commit in the github UI.

Trying.

Thanks!

**Comment by johnynek** [comment]
yeah, that worked: https://github.com/bazelbuild/rules_scala/commit/ecbe5e7833651875810b4bad803430b06d2706ff

**Comment by ittaiz** [comment]
Glad to hear. Are you saying you chose `squash` in the github UI and it was smart enough to only squash the last six commits? Didn't know that.
Also, WDYT about **B** from above?

**Comment by johnynek** [comment]
@ittaiz I agree: #288

**Comment by ittaiz** [comment]
Great. I might be confusing things here but can we use the zipper to
replace the JarCreator used by ScalacProcessor?

**Comment by johnynek** [comment]
@ittaiz I don't think we can because each action needs to make one output. We use JarCreator to produce single outputs from actions, but maybe I'm wrong. Maybe the action can write to a directory, and a subsequent action can read that directory.

That said, I don't see the big win there. That java code was also from bazel, it works and can be compiled with JIT so it is fast. I think spinning up a jvm to make an empty jar is definitely a bummer.

**Comment by ittaiz** [comment]
You're right on both counts. I agree.

**Pull request merged**